docs/vulnerability_matching.md

```diff
@@ -0,0 +1,149 @@
+# Vulnerability Matching
+
+This is a high level overview of the interfaces that must be implemented for end-to-end vulnerability matching.  
+A description of the end-to-end process is described as well.  
+
+## Scanners
+
+Several Scanner interface types exist for extracting contents from container layers.  
+
+### PackageScanner
+
+`claircore.internal.indexer.PackageScanner`  
```

Comment:
These shouldn't be dotted, they should be in go notation like `internal/indexer.PackageScanner`